src/main/scala/com/gu/invoicing/refund/Program.scala

```diff
@@ -33,10 +33,13 @@ object Program {
     }
     System.out.println("Successfully got InvoiceItems")
 
-    val (invoiceId, invoice, items) = decideRelevantInvoice(invoices, itemsByInvoiceId) tap { case (_, invoice, _) =>
-      s"$invoice should be at least posted and not negative" assert (invoice.Amount > 0.0 && invoice.Status == "Posted")
+    val (invoiceId, invoice, items) = decideRelevantInvoice(refund, invoices, itemsByInvoiceId) tap {
+      case (_, invoice, _) =>
+        s"$invoice should be posted and have an amount >= the refund value" assert (
+          invoice.Amount >= refund && invoice.Status == "Posted"
```

Comment:
I guess this is some kind of postcondition based programming? as it's asserting the logic in the function anyway.

Reply:
found an interesting doc, which includes a section on that https://github.com/guardian/invoicing-api/blob/main/adr/lightweight-architecture.md#testing-in-production-via-preconditions-program-postconditions-instead-of-mocked-test